driver/restdriver.go

```diff
@@ -37,19 +52,422 @@ func (driver *RestDriver) Initialize(logger logger.LoggingClient, asyncValues ch
+		resourceName := req.DeviceResourceName
+
+		service := sdk.RunningService()
+
+		deviceResource, ok := service.DeviceResource(deviceName, resourceName)
+		if !ok {
+			driver.logger.Errorf("Incoming Writing ignored. Resource '%s' not found", resourceName)
+			return nil
+		}
+
+		// Now get the query parameters received in the request.
+		// These needs to be sent to the end device
+		// Query parameters needs to be converted to string to append to the uri
+		driver.logger.Debugf("URLRawQuery :%v", req.Attributes["urlRawQuery"])
+		reqParam := fmt.Sprint(req.Attributes["urlRawQuery"])
+
+		// Form URI from the end device parameters and request parameters and
+		// query parameters. Omit uri prefix if it is empty
+		if uriPrefix != "" {
+			uri = "http://" + address + ":" + port + "/" + uriPrefix + "/" + req.DeviceResourceName + "?" + reqParam + ""
+		} else {
+			uri = "http://" + address + ":" + port + "/" + req.DeviceResourceName + "?" + reqParam + ""
+		}
+
+		// Its time to form payload to be sent to end device.
+		// For this fisrt get the data received in the write command request
+		// This data is validated against the expected value type of device resource
+		// With the data and uri create new http PUT request
+		// And, set the content type header for the PUT request
+		driver.logger.Debugf("params = %v", params[i])
+		reading := params[i].Value
+		valueType := deviceResource.Properties.ValueType
+		switch valueType {
+		case common.ValueTypeObject:
+			buf, _ := json.Marshal(reading)
+			driver.logger.Debugf("body = %v", buf)
+			if !json.Valid([]byte(buf)) {
+				driver.logger.Debugf("Invalid JSON string")
+			}
+
+			// Create new PUT request, this will not send request to end device
+			request, err = http.NewRequest(http.MethodPut, uri, bytes.NewReader(buf))
+			if err != nil {
+				// handle error
+				return fmt.Errorf("PUT request creation failed")
+			}
+			// Set content type as application/json
+			request.Header.Set("Content-Type", "application/json; charset=UTF-8")
+
+		case common.ValueTypeBool, common.ValueTypeString, common.ValueTypeUint8,
+			common.ValueTypeUint16, common.ValueTypeUint32, common.ValueTypeUint64,
+			common.ValueTypeInt8, common.ValueTypeInt16, common.ValueTypeInt32,
+			common.ValueTypeInt64, common.ValueTypeFloat32, common.ValueTypeFloat64:
+			// All other types
+			contentType := "text/plain"
+			_, err = driver.validateData(deviceResource, reading, deviceResource.Properties.ValueType, contentType)
+			if err != nil {
+				// handle error
+				return fmt.Errorf("PUT request data is not valid")
+			}
+			// Create new PUT request
+			request, err = http.NewRequest(http.MethodPut, uri, strings.NewReader(cast.ToString(reading)))
+			if err != nil {
+				// handle error
+				return fmt.Errorf("PUT request creation failed")
+			}
+			// Set content type as text/plain
+			request.Header.Set("Content-Type", "text/plain; charset=UTF-8")
```

Comment:
use 
https://github.com/edgexfoundry/go-mod-core-contracts/blob/7e0c83d7b629306e64997ecd877af1515e598311/common/constants.go#L289
https://github.com/edgexfoundry/go-mod-core-contracts/blob/7e0c83d7b629306e64997ecd877af1515e598311/common/constants.go#L295

Reply:
Using the constants as suggested